src/components/utils/src/convert_utils.cc

```diff
@@ -70,3 +70,9 @@ uint64_t utils::ConvertLongLongUIntToUInt64(
 
   return static_cast<uint64_t>(value);
 }
+
+int64_t utils::Int64DivisionRoundUp(const int64_t numerator,
+                                    const uint64_t denominator) {
```

Comment:
@OHerasym mixing arithmetic of signed and unsigned in general is not good idea in C/C++. Small negative numerators will be converted to big unsigned positive numbers. Try numerator -100 and denominator 20. What do you expect? -5? Probably it will be 922337203685477575.
I suggest to store the sign and then do the calculations by absolute values in unsigned format then restore the sign in the final result. OR use uint32_t for unsigned (keep signed bigger).